training/src/main/scala/com/airbnb/aerosolve/training/pipeline/GenericPipeline.scala

```diff
@@ -470,7 +473,7 @@ object GenericPipeline {
     } else if (isMulticlass) {
       Evaluation.evaluateMulticlassClassification(records)
     } else {
-      Evaluation.evaluateBinaryClassification(records, bins, metric)
+      Evaluation.evaluateBinaryClassificationWithResults(records, bins, metric, resultsOutputPath)
```

Comment:
I got rid of `appendMetricAndResult`. Now I am just relying on the metrics buffer and `resultsOutputPath` is only being used in one place here `Evaluation#L78`: 
```
    if (resultsOutputPath != null) {
      ResultUtil.writeResults(resultsOutputPath, metrics.toArray, holdThresholdPrecisionRecall.toArray, trainThresholdPrecisionRecall.toArray)
    }
```